swagger.yaml

```diff
@@ -0,0 +1,1358 @@
+openapi: 3.0.1
+info:
+  title: Ably REST API
+  description: API specification for Ably
+  version: "1.1.0"
+components:
+  securitySchemes:
+    basicAuth:
+      type: http
+      scheme: basic
+    bearerAuth:
+      type: http
+      scheme: bearer
+      bearerFormat: JWT
+
+  headers:
+    # TODO: Issue exists for integrating pagination with OpenAPI, https://github.com/OAI/OpenAPI-Specification/issues/667. If this is ever actioned, we should enable it within our spec.
+    Link:
+      description: Links to related resources, in the format defined by [RFC 5988](https://tools.ietf.org/html/rfc5988#section-5). This will potentially include a link with relation type `next`, `first`, and `current`, where appropiate.
+      schema:
+        type: string
+        pattern: '(<(.*)?>; rel=\"(first|current|last)?\",)*(<(.*)?>; rel=\"(first|current|last)?\")+' 
+      required: true
+      
+    ServerId:
+      description: The ID for the server communicated with.
+      schema:
+        type: string
+      required: true
+      
+    ErrorCode:
+      description: The error code.
+      schema:
+        type: string
+    
+    ErrorMessage:
+      description: The error message.
+      schema:
+        type: string
+
+  parameters:
+    # Currently each parameter needs to be defined seperately. If https://github.com/OAI/OpenAPI-Specification/issues/445 ever goes through, can amalagmate pagination parameters
+    paginationStart:
+      name: start
+      in: query
+      schema:
+        type: string
+        default: '0'
```

Comment:
Do you happen to know what the default value is? I know they can be strings (such as `now`), thus why I went with string, but may be wrong applying it to `start`, and it may only apply to `end`.

Reply:
We should have no default.